core/trino-spi/src/main/java/io/trino/spi/eventlistener/ClauseInfo.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.eventlistener;
+
+public enum ClauseInfo
+{
+    SELECT,
+    FROM,
+    WHERE,
+    GROUP_BY,
+    HAVING,
+    WINDOW,
+    ORDER_BY,
+    OFFSET,
+    LIMIT,
+    JOIN,
+    UNNEST,
+    UPDATE,
+    MERGE,
+    ROW_FILTER,
+    COLUMN_MASK,
+    VALUES,
+    PATTERN_RECOGNITION,
+    UNSPECIFIED
```

Comment:
This is used

- in CallTask -- add explicit CALL entry here
- in the deprecated RoutineInfo constructor, so let's deprecate the UNSPECIFIED as well (or maybe we can remove it, see below)

Reply:
Removed